src/lib/app-state-hoc.jsx

```diff
@@ -12,25 +14,37 @@ const enhancer = composeEnhancers(
         throttle(300, {leading: true, trailing: true})
     )
 );
-const store = createStore(reducer, intlInitialState, enhancer);
-
-import ErrorBoundary from '../containers/error-boundary.jsx';
 
 /*
  * Higher Order Component to provide redux state
  * @param {React.Component} WrappedComponent - component to provide state for
  * @returns {React.Component} component with redux and intl state provided
  */
 const AppStateHOC = function (WrappedComponent) {
-    const AppStateWrapper = ({...props}) => (
-        <Provider store={store}>
-            <IntlProvider>
-                <ErrorBoundary>
-                    <WrappedComponent {...props} />
-                </ErrorBoundary>
-            </IntlProvider>
-        </Provider>
-    );
+    class AppStateWrapper extends React.Component {
+        constructor (props) {
+            super(props);
+            this.store = createStore(reducer, (props.intl || intlInitialState), enhancer);
```

Comment:
We should add some explanation about how this works.  Maybe in the doc comment? 
```javascript
 /*
  * Higher Order Component to provide redux state.  If an `intl` prop is provided, it will override the internal
  * `intl` redux state.
  * @param {React.Component} WrappedComponent - component to provide state for
  * @returns {React.Component} component with redux and intl state provided
  */
```